Sources/MessagingInApp/Gist/Managers/MessageQueueManager.swift

```diff
@@ -72,21 +97,16 @@ class MessageQueueManager {
             Logger.instance.info(message: "Checking Gist queue service")
             if let userToken = UserManager().getUserToken() {
                 QueueManager(siteId: Gist.shared.siteId, dataCenter: Gist.shared.dataCenter)
-                    .fetchUserQueue(userToken: userToken, completionHandler: { response in
+                    .fetchUserQueue(userToken: userToken, completionHandler: { [weak self] response in
                         switch response {
                         case .success(nil):
                             Logger.instance.info(message: "No changes to remote queue")
                         case .success(let responses):
                             guard let responses else {
                                 return
                             }
-                            // To prevent us from showing expired / revoked messages, clear user messages from local queue.
-                            self.clearUserMessagesFromLocalStore()
-                            Logger.instance.info(message: "Gist queue service found \(responses.count) new messages")
-                            for queueMessage in responses {
-                                let message = queueMessage.toMessage()
-                                self.handleMessage(message: message)
-                            }
```

Comment:
I moved this block into a class function so I could write automated tests against the logic.